core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -212,7 +226,36 @@ static Permissions createPermissions(Environment environment) throws IOException
             // we just need permission to remove the file if its elsewhere.
             policy.add(new FilePermission(environment.pidFile().toString(), "delete"));
         }
-        return policy;
+    }
+    
+    static void addBindPermissions(Permissions policy, Settings settings) throws IOException {
+        // http is simple
+        String httpRange = settings.get("http.netty.port", 
+                               settings.get("http.port", 
+                                       NettyHttpServerTransport.DEFAULT_PORT_RANGE));
+        policy.add(new SocketPermission("localhost:" + httpRange, "listen,resolve"));
+        // transport is waaaay overengineered
+        Map<String, Settings> profiles = settings.getGroups("transport.profiles", true);
```

Comment:
> PS. I presume that just accessing a static const (like `NettyTransport.DEFAULT_PORT_RANGE` ) doesn't execute static coded?

Correct. As long as the variable is marked `final`, is a primitive or a `String`, and is initialized by a compile-time constant expression then accessing it will not cause the class to be considered accessed and therefore not trigger the class variable initializers and static initializers to run ([JLS12.4.1](https://docs.oracle.com/javase/specs/jls/se7/html/jls-12.html#jls-12.4.1), [JLS12.4.2](https://docs.oracle.com/javase/specs/jls/se7/html/jls-12.html#jls-12.4.2)). So in this case,

```java
public static final String DEFAULT_PORT_RANGE = "9300-9400";
```

is okay and accessing it would not trigger `NettyUtils#setup` in the static initializer.

However, this strikes me as very fragile and subtle and I would want a minimum a test to ensure this behavior is preserved.